src/main/java/de/retest/recheck/ui/descriptors/idproviders/DefaultRetestIdProvider.java

```diff
@@ -2,49 +2,59 @@
 
 import static de.retest.recheck.util.RetestIdUtil.cutTypeQualifier;
 import static de.retest.recheck.util.RetestIdUtil.normalizeAndCut;
+import static java.util.UUID.randomUUID;
 
-import java.util.HashSet;
-import java.util.Set;
-import java.util.UUID;
+import java.util.HashMap;
+import java.util.Map;
 
 import org.apache.commons.lang3.StringUtils;
 
 import de.retest.recheck.ui.descriptors.IdentifyingAttributes;
 
+/**
+ * Use the first non-null, non-blank of native id attribute, text or type to create a unique retestId. If the id, text
+ * or type already exists, use a random suffix to make it unique.
+ */
 public class DefaultRetestIdProvider implements RetestIdProvider {
```

Comment:
Hmm. Actually we should do both. I'll rename it and have the Default inherit from that. We should move or rename the DefaultRetestIdProvider, as this would be a breaking change...